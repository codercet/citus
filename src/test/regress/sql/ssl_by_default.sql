-- Citus uses ssl by default now. It does so by turning on ssl and if needed generate self
-- signed certificates.

-- To test this we will verify that SSL is set to ON for all machines, and we will make
-- sure connections to workers use SSL by having it required in citus.conn_nodeinfo and
-- lastly we will inspect the ssl state for connections to the workers

-- we expect all nodes to have ssl on
SHOW ssl;
SELECT run_command_on_workers($$
    SHOW ssl;
$$);

-- we expect all nodes to have node_conninfo set to 'sslmode=require'
SHOW citus.node_conninfo;
SELECT run_command_on_workers($$
    SHOW citus.node_conninfo;
$$);

-- we expect all connections to the workers have ssl on
SELECT run_command_on_workers($$
    SELECT ssl FROM pg_stat_ssl WHERE pid = pg_backend_pid();
$$);
