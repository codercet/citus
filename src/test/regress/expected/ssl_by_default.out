-- Citus uses ssl by default now. It does so by turning on ssl and if needed generate self
-- signed certificates.
-- To test this we will verify that SSL is set to ON for all machines, and we will make
-- sure connections to workers use SSL by having it required in citus.conn_nodeinfo and
-- lastly we will inspect the ssl state for connections to the workers
-- we only enable ssl during our tests on 10 and above due to a required restart in pg 9
SHOW server_version \gset
SELECT substring(:'server_version', '\d+')::int >= 10 AS version_ten_or_above;
 version_ten_or_above 
----------------------
 t
(1 row)

-- we expect all nodes to have ssl on
SHOW ssl;
 ssl 
-----
 on
(1 row)

SELECT run_command_on_workers($$
    SHOW ssl;
$$);
 run_command_on_workers 
------------------------
 (localhost,57637,t,on)
 (localhost,57638,t,on)
(2 rows)

-- we expect all nodes to have node_conninfo set to 'sslmode=require'
SHOW citus.node_conninfo;
 citus.node_conninfo 
---------------------
 sslmode=require
(1 row)

SELECT run_command_on_workers($$
    SHOW citus.node_conninfo;
$$);
       run_command_on_workers        
-------------------------------------
 (localhost,57637,t,sslmode=require)
 (localhost,57638,t,sslmode=require)
(2 rows)

-- we expect all connections to the workers have ssl on
SELECT run_command_on_workers($$
    SELECT ssl FROM pg_stat_ssl WHERE pid = pg_backend_pid();
$$);
 run_command_on_workers 
------------------------
 (localhost,57637,t,t)
 (localhost,57638,t,t)
(2 rows)

